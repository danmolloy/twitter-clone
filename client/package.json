{
  "name": "twitter-clone",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@apollo/client": "^3.4.10",
    "@craco/craco": "^6.2.0",
    "@heroicons/react": "^1.0.4",
    "@material-ui/core": "^4.12.3",
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.3.2",
    "@testing-library/user-event": "^7.1.2",
    "@types/jest": "^24.0.0",
    "@types/node": "^12.0.0",
    "@types/react": "^16.9.0",
    "@types/react-dom": "^16.9.0",
    "@types/react-redux": "^7.1.7",
    "@types/react-router": "^5.1.16",
    "@types/react-router-dom": "^5.1.8",
    "faker": "^5.5.3",
    "graphql": "^15.5.2",
    "nanoid": "^3.1.25",
    "pretty": "^2.0.0",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-router-dom": "^5.2.0",
    "react-scripts": "4.0.3",
    "typescript": "~4.1.5"
  },
  "scripts": {
    "start": "craco start",
    "build": "craco build",
    "test": "craco test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@types/faker": "^5.5.8",
    "@types/pretty": "^2.0.0",
    "autoprefixer": "^9.8.6",
    "postcss": "^7.0.36",
    "prettier": "^2.3.2",
    "tailwindcss": "npm:@tailwindcss/postcss7-compat@^2.2.7"
  }
}
