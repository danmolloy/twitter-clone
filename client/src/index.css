@tailwind base;
@tailwind components;
@tailwind utilities;

.side-icon {
  @apply h-12 w-auto mx-8 sm:mx-2 rounded-full p-2 hover:bg-gray-200;
}

.lg-side-icon {
  @apply hover:bg-gray-200 flex flex-row items-center md:ml-6 md:pr-4 my-1 rounded-full;
}

.sidebar-text {
  @apply hidden md:flex text-lg;
}

.compose-tweet-icons {
  @apply w-10 h-auto twitter-blue p-2 hover:bg-blue-50 rounded-full;
}

.tweet-options {
  @apply w-10 h-auto p-2 rounded-full;
}

.tweet-btn {
  @apply h-8 bg-twitter-blue text-white px-3 my-3 rounded-full font-bold mr-3 shadow;
}

.tweet-btn-disabled {
  @apply h-8 bg-blue-opacity text-white px-3 my-3 rounded-full font-bold mr-3;
}

.r-bar-article {
  @apply p-2 w-full;
}

.welcome-background {
  background-image: linear-gradient(to bottom right, #ffffff, #1da1f2);
}


.blue-hover {
  background-color: #1DA1F2;
  opacity: .5;
}

.bg-twitter-blue {
  background-color: #1DA1F2;
}

.twitter-blue {
  --tw-text-opacity: 1;
  color: #1DA1F2;
}


.bg-blue-opacity {
  background-color: #1DA1F2;
  opacity: .5;
}


svg path {
  stroke-width: 1.5;
}

.side-bar {
  @apply self-end fixed flex flex-row sm:flex-col sm:mt-0 ml-0 w-full 
  sm:w-24 md:w-60 sm:h-screen border-t sm:border-t-0 sm:border-r 
  border-gray-200 md:items-start justify-center sm:justify-start items-center bg-white;
}

/* Home */
.home-header {
  @apply border-b border-gray-200 h-14 flex flex-row justify-between;
}

.h2 {
  @apply text-xl font-semibold;
}

/* Notifications */
.notification-btn-selected {
  @apply font-semibold w-1/2 h-12 hover:bg-gray-200 flex flex-col items-center justify-between pt-2;
}

.notification-btn {
  @apply  w-1/2 h-12 hover:bg-gray-200 flex flex-col items-center justify-between pt-2 text-gray-500;
}

.tab-line {
  @apply border bg-twitter-blue h-1.5 rounded-full w-12;
}

/* Profile*/

.selected-tweet-filter {
  @apply w-1/2 border-b font-semibold hover:bg-gray-200 flex flex-col justify-between items-center pt-2;
}

.deselected-tweet-filter {
  @apply w-1/2 hover:bg-gray-200 border-b text-center py-2;
}

.follow-btn {
  @apply font-bold bg-black text-white rounded-full py-2 px-4 my-8 mr-8;
}

.unfollow-btn {
  @apply font-bold border border-gray-300 rounded-full py-2 px-4 my-8 mr-8;
}

textarea {
  resize: none
}