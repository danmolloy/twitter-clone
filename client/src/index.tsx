import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import App from './Components/App/App';
import * as serviceWorker from './serviceWorker';
import {ApolloClient, InMemoryCache, ApolloProvider, createHttpLink} from '@apollo/client';
import { setContext } from '@apollo/client/link/context';
import { AUTH_TOKEN } from './constants';

const httpLink = createHttpLink({
  uri: 'https://twitr-clone.herokuapp.com/'
});

const authLink = setContext((_, { headers }) => {
  const token = localStorage.getItem(AUTH_TOKEN);
  return {
    headers: {
      ...headers,
      authorization: token ? `Bearer ${token}` : ''
    }
  };
});

const client = new ApolloClient({
  link: authLink.concat(httpLink),
  cache: new InMemoryCache()
});

ReactDOM.render(
  <React.StrictMode>
    <ApolloProvider client={client}>
      <App />
    </ApolloProvider>
  </React.StrictMode>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: https://bit.ly/CRA-PWA
serviceWorker.unregister();
