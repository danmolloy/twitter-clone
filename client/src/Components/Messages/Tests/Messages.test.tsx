import { act } from "react-dom/test-utils";
import { unmountComponentAtNode, render } from "react-dom";
import { MemoryRouter } from "react-router";
import { MockedProvider } from "@apollo/client/testing";
import { AUTH_TOKEN } from "../../../constants";
import pretty from "pretty";
import { fireEvent, getByText } from "@testing-library/dom";
import { MessagesMock } from "./MessagesMock"

let container: any = null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
  global.localStorage.setItem(AUTH_TOKEN, "auth-token");
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  global.localStorage.removeItem(AUTH_TOKEN);
});