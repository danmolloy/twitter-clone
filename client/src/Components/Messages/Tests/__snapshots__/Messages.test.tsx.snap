// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat preview click redirects to chat 1`] = `
"<div>
  <div id=\\"main-content\\" class=\\"flex flex-row w-screen h-screen justify-start\\">
    <div id=\\"side-bar\\" class=\\"side-bar\\"><a class=\\"md:ml-2\\" id=\\"twitter-home-link\\" title=\\"Home\\" href=\\"/home\\"><img src=\\"/favico.ico\\" alt=\\"Blue Twitter bird\\" class=\\"hidden sm:flex w-12 h-auto mx-6 my-4 p-2 rounded hover:bg-blue-50\\"></a><a class=\\"lg-side-icon\\" id=\\"home-link\\" title=\\"Home\\" href=\\"/home\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"side-icon\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6\\"></path>
        </svg><label for=\\"home-link\\" class=\\"sidebar-text\\">Home</label></a><a title=\\"Explore\\" class=\\"lg-side-icon\\" id=\\"explore-link\\" href=\\"/explore\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"side-icon\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M7 20l4-16m2 16l4-16M6 9h14M4 15h14\\"></path>
        </svg><label for=\\"explore-link\\" class=\\"sidebar-text\\">Explore</label></a><a title=\\"Notifications\\" class=\\"lg-side-icon\\" id=\\"notifications-link\\" href=\\"/notifications\\">
        <div class=\\"bg-blue-600 h-2 w-2 rounded-full z-10 -mr-8 sm:-mr-2 -mt-6 sm:-mt-0\\"></div><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"side-icon\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M15 17h5l-1.405-1.405A2.032 2.032 0 0118 14.158V11a6.002 6.002 0 00-4-5.659V5a2 2 0 10-4 0v.341C7.67 6.165 6 8.388 6 11v3.159c0 .538-.214 1.055-.595 1.436L4 17h5m6 0v1a3 3 0 11-6 0v-1m6 0H9\\"></path>
        </svg><label for=\\"notifications-link\\" class=\\"sidebar-text\\">Notifications</label>
      </a><a title=\\"Messages\\" class=\\"lg-side-icon\\" href=\\"/messages\\">
        <div id=\\"unread-msg-alert\\" class=\\"bg-blue-600 h-2 w-2 rounded-full z-10 -mr-8 sm:-mr-2 -mt-6 sm:-mt-0\\"></div><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"side-icon\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M20 13V6a2 2 0 00-2-2H6a2 2 0 00-2 2v7m16 0v5a2 2 0 01-2 2H6a2 2 0 01-2-2v-5m16 0h-2.586a1 1 0 00-.707.293l-2.414 2.414a1 1 0 01-.707.293h-3.172a1 1 0 01-.707-.293l-2.414-2.414A1 1 0 006.586 13H4\\"></path>
        </svg><label for=\\"messages-link\\" class=\\"sidebar-text\\">Messages</label>
      </a><button class=\\"sm:hidden lg-side-icon\\" title=\\"User options\\"><img alt=\\"Current user\\" src=\\"profilePic.jpg\\" class=\\"side-icon rounded-full\\"></button><a id=\\"profile-link\\" title=\\"Profile\\" class=\\"hidden sm:flex lg-side-icon\\" href=\\"/leGuin\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"side-icon hidden sm:flex\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z\\"></path>
        </svg><label for=\\"profile-link\\" class=\\"sidebar-text\\">Profile</label></a><a title=\\"Sign out\\" id=\\"sign-out-btn\\" class=\\"hidden sm:flex lg-side-icon\\" href=\\"/home\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"side-icon hidden sm:flex\\" id=\\"more-icon\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M17 16l4-4m0 0l-4-4m4 4H7m6 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h4a3 3 0 013 3v1\\"></path>
        </svg><label for=\\"sign-out\\" class=\\"sidebar-text\\">Sign Out</label></a></div>
    <div class=\\"sm:ml-24 md:ml-60 mb-0 w-full max-w-2xl sm:mr-2 md:mr-0\\">
      <div class=\\"border-r min-h-full w-full flex flex-col justify-between \\">
        <div id=\\"chat-header\\" class=\\"border-b border-r p-2 flex flex-row items-center z-10 bg-white w-full fixed max-w-2xl\\"><a href=\\"/messages\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" id=\\"messages-link\\" class=\\"w-10 hover:bg-gray-100 rounded p-1\\">
              <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M10 19l-7-7m0 0l7-7m-7 7h18\\"></path>
            </svg></a><button class=\\"flex flex-row items-center hover:bg-gray-100 rounded justify-center\\"><img src=\\"/dan.jpg\\" class=\\"rounded-full w-12 h-12 ml-4\\">
            <h3 class=\\"mx-4 font-semibold\\">Dan MolloyUrsula LeGuin</h3>
          </button></div>
        <div id=\\"chat-window\\" class=\\" mt-16 w-full mb-16\\">
          <div class=\\"flex flex-row m-1  items-center\\"><img src=\\"/dan.jpg\\" class=\\"rounded-full w-12 h-12\\">
            <div class=\\"py-1 px-2 m-4 max-w-xs rounded-md bg-gray-100\\">
              <p>Hi @leGuin! Welcome to Twitter. To get you started, we are following each other. You can unfollow me by navigating to my profile.</p>
            </div>
          </div>
          <div class=\\"flex flex-row m-1  items-center\\"><img src=\\"/dan.jpg\\" class=\\"rounded-full w-12 h-12\\">
            <div class=\\"py-1 px-2 m-4 max-w-xs rounded-md bg-gray-100\\">
              <p>Find other users to follow in Explore. You can comment, like and retweet their posts, as well as message them. We have some interesting users here!</p>
            </div>
          </div>
          <div class=\\"flex flex-row m-1  items-center\\"><img src=\\"/dan.jpg\\" class=\\"rounded-full w-12 h-12\\">
            <div class=\\"py-1 px-2 m-4 max-w-xs rounded-md bg-gray-100\\">
              <p>Check out your profile - you can view your posts, retweets, and edit some of your user details.</p>
            </div>
          </div>
          <div class=\\"flex flex-row m-1  items-center\\"><img src=\\"/dan.jpg\\" class=\\"rounded-full w-12 h-12\\">
            <div class=\\"py-1 px-2 m-4 max-w-xs rounded-md bg-gray-100\\">
              <p>Once finished, you can sign out. Soon afterwards your account will be automatically deleted. Cheers, Dan</p>
            </div>
          </div>
        </div>
        <div class=\\"flex flex-row justify-center items-center mb-16 bottom-0 sm:mb-0 flex border-t border-r bg-white z-10 fixed w-full md:-ml-0 max-w-2xl\\"><input class=\\"border rounded-full m-4 p-2\\" placeholder=\\"hello\\" value=\\"\\"><button class=\\"tweet-btn\\">Send</button></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`shows chat previews 1`] = `
"<div>
  <div id=\\"main-content\\" class=\\"flex flex-row w-screen h-screen justify-start\\">
    <div id=\\"side-bar\\" class=\\"side-bar\\"><a class=\\"md:ml-2\\" id=\\"twitter-home-link\\" title=\\"Home\\" href=\\"/home\\"><img src=\\"/favico.ico\\" alt=\\"Blue Twitter bird\\" class=\\"hidden sm:flex w-12 h-auto mx-6 my-4 p-2 rounded hover:bg-blue-50\\"></a><a class=\\"lg-side-icon\\" id=\\"home-link\\" title=\\"Home\\" href=\\"/home\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"side-icon\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6\\"></path>
        </svg><label for=\\"home-link\\" class=\\"sidebar-text\\">Home</label></a><a title=\\"Explore\\" class=\\"lg-side-icon\\" id=\\"explore-link\\" href=\\"/explore\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"side-icon\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M7 20l4-16m2 16l4-16M6 9h14M4 15h14\\"></path>
        </svg><label for=\\"explore-link\\" class=\\"sidebar-text\\">Explore</label></a><a title=\\"Notifications\\" class=\\"lg-side-icon\\" id=\\"notifications-link\\" href=\\"/notifications\\">
        <div class=\\"bg-blue-600 h-2 w-2 rounded-full z-10 -mr-8 sm:-mr-2 -mt-6 sm:-mt-0\\"></div><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"side-icon\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M15 17h5l-1.405-1.405A2.032 2.032 0 0118 14.158V11a6.002 6.002 0 00-4-5.659V5a2 2 0 10-4 0v.341C7.67 6.165 6 8.388 6 11v3.159c0 .538-.214 1.055-.595 1.436L4 17h5m6 0v1a3 3 0 11-6 0v-1m6 0H9\\"></path>
        </svg><label for=\\"notifications-link\\" class=\\"sidebar-text\\">Notifications</label>
      </a><a title=\\"Messages\\" class=\\"lg-side-icon\\" href=\\"/messages\\">
        <div id=\\"unread-msg-alert\\" class=\\"bg-blue-600 h-2 w-2 rounded-full z-10 -mr-8 sm:-mr-2 -mt-6 sm:-mt-0\\"></div><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"side-icon\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M20 13V6a2 2 0 00-2-2H6a2 2 0 00-2 2v7m16 0v5a2 2 0 01-2 2H6a2 2 0 01-2-2v-5m16 0h-2.586a1 1 0 00-.707.293l-2.414 2.414a1 1 0 01-.707.293h-3.172a1 1 0 01-.707-.293l-2.414-2.414A1 1 0 006.586 13H4\\"></path>
        </svg><label for=\\"messages-link\\" class=\\"sidebar-text\\">Messages</label>
      </a><button class=\\"sm:hidden lg-side-icon\\" title=\\"User options\\"><img alt=\\"Current user\\" src=\\"profilePic.jpg\\" class=\\"side-icon rounded-full\\"></button><a id=\\"profile-link\\" title=\\"Profile\\" class=\\"hidden sm:flex lg-side-icon\\" href=\\"/leGuin\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"side-icon hidden sm:flex\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z\\"></path>
        </svg><label for=\\"profile-link\\" class=\\"sidebar-text\\">Profile</label></a><a title=\\"Sign out\\" id=\\"sign-out-btn\\" class=\\"hidden sm:flex lg-side-icon\\" href=\\"/home\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"side-icon hidden sm:flex\\" id=\\"more-icon\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M17 16l4-4m0 0l-4-4m4 4H7m6 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h4a3 3 0 013 3v1\\"></path>
        </svg><label for=\\"sign-out\\" class=\\"sidebar-text\\">Sign Out</label></a></div>
    <div class=\\"sm:ml-24 md:ml-60 mb-0 w-full max-w-2xl sm:mr-2 md:mr-0\\">
      <div class=\\"border-b flex flex-row justify-between border-r\\">
        <h2 class=\\"text-xl font-semibold pl-3 pt-3 pb-3\\">Messages</h2>
        <div class=\\"flex flex-row\\"><button class=\\"search-following\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"w-9 p1 h-auto hover:bg-gray-300 my-2 rounded-full mr-3\\">
              <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M12 6v6m0 0v6m0-6h6m-6 0H6\\"></path>
            </svg></button></div>
      </div>
      <div class=\\"flex flex-col items-center w-full border-r min-h-screen\\"><a class=\\"chat-preview border-b flex flex-row p-4 justify-between hover:bg-gray-100 w-full\\" href=\\"/messages/395ed01a-5e12-4220-b94e-ab0bc76a691c\\">
          <div class=\\"flex flex-row\\"><img src=\\"/dan.jpg\\" class=\\"rounded-full w-12\\">
            <div class=\\"flex flex-col ml-4\\">
              <h3 class=\\"font-semibold\\">Dan Molloy</h3>
              <div class=\\"flex flex-row\\">
                <p id=\\"msg-text-preview\\" class=\\"font-bold\\">@danmolloy:
                  Once finished, you c<span class=\\"text-gray-400\\"> ...</span></p>
              </div>
            </div>
          </div>
          <p class=\\"text-gray-500\\">Wed Nov 24 2021 11:03</p>
        </a></div>
    </div>
  </div>
</div>"
`;
